tools/deb/istio-start.sh

```diff
@@ -54,7 +62,7 @@ fi
 ${ISTIO_BIN_BASE}/istio-iptables.sh
 
 if [ -f ${ISTIO_BIN_BASE}/pilot-agent ]; then
-  exec su -s /bin/bash -c "INSTANCE_IP=${ISTIO_SVC_IP} POD_NAME=${POD_NAME} POD_NAMESPACE=${NS} exec ${ISTIO_BIN_BASE}/pilot-agent proxy --serviceCluster $SVC > ${ISTIO_LOG_DIR}/istio.log" istio-proxy
+  exec su -s /bin/bash -c "INSTANCE_IP=${ISTIO_SVC_IP} POD_NAME=${POD_NAME} POD_NAMESPACE=${NS} exec ${ISTIO_BIN_BASE}/pilot-agent proxy --serviceCluster $SVC --discoveryAddress istio-pilot.${ISTIO_SYSTEM_NAMESPACE} --controlPlaneAuthPolicy $CONTROL_PLANE_AUTH_POLICY > ${ISTIO_LOG_DIR}/istio.log" istio-proxy
```

Comment:
what would happen if --discoveryAddress is not passed?

Reply:
The namespace part of the discovery address is used by pilot-agent to construct spiffe uri to authenticate. So it needs to be correct otherwise the client will reject the secure connection. The default value works for plain connections.